registration.go

```diff
@@ -126,6 +127,42 @@ func (d *Distro) Unregister() (err error) {
 	return d.backend.WslUnregisterDistribution(d.Name())
 }
 
+// Uninstall removes the distro's associated AppxPackage (if there is one)
+// and unregisters the distro.
+func (d *Distro) Uninstall(ctx context.Context) (err error) {
+	defer decorate.OnError(&err, "Distro %q uninstall", d.name)
+
+	defer func() {
+		// If no error is returned, we unregister
+		if err != nil {
+			return
+		}
+		err = d.backend.WslUnregisterDistribution(d.Name())
```

Comment:
I find it quite backward to unregister the distro after the appx was removed.

I understand you are doing because you need to get some info from the registry first, but what if we insert the unregister just after the PackageFamilyName lookup, before the error check. WDYT?

(I find an "important" operation, which is not recovery or tear down code, being in the `defer` call a little bit as a bad pattern)

Reply:
Unregistering the distro erases irrecoverable information, the AppxFamilyName in particular. If the Appx uninstall fails, I want to preserve this info. In that way, we only unregister if we succeeded to uninstall, and otherwise it can be tried again later.

I agree to move it outside of the defer statement.